packages/block/src/header.ts

```diff
@@ -342,7 +342,7 @@ export class BlockHeader {
     }
 
     // Check for constant values for PoS blocks
-    if (this._common.consensusType() === ConsensusType.ProofOfStake) {
+    if (this._common.consensusType() === ConsensusType.ProofOfStake && this.number >= this._common.hardforks().filter((hf: any) => hf.name === 'merge').block) {
```

Comment:
consensusType() changes to ProofOfStake after arriving at the merge hardfork block, so the added condition is redundant

Reply:
Okay, let me try backing this change out and see if so. I think I was seeing it throw here with some PoS issue but that may be cause of me passing the wrong common instance in the test.

Reply:
Yes, this check can (and should) definitely be removed.